app/models.py

```diff
@@ -431,6 +412,35 @@ class AuditEvent(db.Model):
         object_type, object_id
     )
 
+    def __init__(self, audit_type, user, data, db_object):
+        if not AuditTypes.is_valid_audit_type(audit_type):
+            raise ValueError("{} is not a valid audit type".format(audit_type))
+        self.type = audit_type
```

Comment:
Can we add `.value` here instead of on every call? I assume we're only ever going to create events using the `AuditTypes`